src/pages/iv-replenishment/forms/IvReplenishementsForm.js

```diff
@@ -0,0 +1,201 @@
+// ** MUI Imports
+import { Grid } from '@mui/material'
+import { DataSets } from 'src/resources/DataSets'
+
+import * as yup from 'yup'
+import toast from 'react-hot-toast'
+
+import FormShell from 'src/components/Shared/FormShell'
+
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useContext, useEffect, useState, useRef } from 'react'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { IVReplenishementRepository } from 'src/repositories/IVReplenishementRepository'
+import { useInvalidate } from 'src/hooks/resource'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { ControlContext } from 'src/providers/ControlContext'
+import { useForm } from 'src/hooks/form'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import CustomDatePicker from 'src/components/Inputs/CustomDatePicker'
+import { formatDateFromApi } from 'src/lib/date-helper'
+import CustomTextArea from 'src/components/Inputs/CustomTextArea'
+
+const IvReplenishementsForm = ({ labels, maxAccess, setStore, store }) => {
+  const { postRequest, getRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const { recordId } = store
+
+  const invalidate = useInvalidate({
+    endpointId: IVReplenishementRepository.IvReplenishements.qry
+  })
+
+  const { formik } = useForm({
+    maxAccess,
+    initialValues: {
+      recordId: null,
+      siteId: '',
+      dateFrom: null,
+      dateTo: null,
+      date: new Date(),
+      notes: ''
+    },
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      siteId: yup.string().required(),
+      dateFrom: yup.string().required(),
+      dateTo: yup.string().required(),
+      date: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      const recordId = obj.recordId
+
+      const response = await postRequest({
+        extension: IVReplenishementRepository.IvReplenishements.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!recordId) {
+        toast.success(platformLabels.Added)
+        formik.setValues({
+          ...obj,
+          recordId: response.recordId
+        })
+        setStore(prevStore => ({
+          ...prevStore,
+          recordId: response.recordId
+        }))
+      } else toast.success(platformLabels.Edited)
+      invalidate()
+    }
+  })
+  const editMode = !!recordId
```

Comment:
use !formik.values.recordId  instead !!recordId

Reply:
i am using the record Id stored